### DISCUSSION TITLE
Bug? Linux versions don't launch

### DISCUSSION THREAD

**Discussion by Qronikarz** [TagStudioDev/TagStudio#182] [Q&A] [answered]
Don't know if it's a TagStudio bug or the case of Mint 21.3 being ancient and not having modern QT dependencies so figured I'll first ask in discussions.

None of the Linux versions launch for me so I run them from terminal and here are the results.
Portable:
```
userhere@ldesk5:~/Pobrane$ ./tagstudio
[QT DRIVER] Config File not specified, defaulting to /home/userhere/.config/TagStudio/TagStudio.ini
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/tmp/_MEIxxbmk6/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: vnc, eglfs, linuxfb, offscreen, wayland-egl, minimal, xcb, minimalegl, vkkhrdisplay, wayland.

Przerwane (zrzut pamięci)

```
Normal:
```
userhere@ldesk5:~/Pobrane/tagstudio$ ./tagstudio
[QT DRIVER] Config File not specified, defaulting to /home/userhere/.config/TagStudio/TagStudio.ini
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/home/userhere/Pobrane/tagstudio/_internal/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: vnc, eglfs, linuxfb, offscreen, wayland-egl, minimal, xcb, minimalegl, vkkhrdisplay, wayland.

Przerwane (zrzut pamięci)

```

**Comment by CyanVoxel** [comment]
Tagging @xarvex on this

**Comment by xarvex** [comment, marked as answer]
I noticed the line:
```
From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin
```

This leads me to believe the package is missing on your system. Since Mint is Ubuntu-based, I believe this should be `apt install libxcb-cursor0`, if not then do a search and see what it might be called in your repositories.

Once you've done that, try running the binary again.

**Reply by Qronikarz** [reply]
Yes, that worked. However there should be info in the program that you lack the dependency. For example when installing .deb it opens a software manager window that tells you what dependencies will be installed. Also maybe it could be good to mention it somewhere in the readme.

**Reply by xarvex** [reply]
> Yes, that worked. However there should be info in the program that you lack the dependency. For example when installing .deb it opens a software manager window that tells you what dependencies will be installed. Also maybe it could be good to mention it somewhere in the readme.

Really we didn't know that this was a required dependency, this is the first binary distributed and this package is not something that appears in `ldd`, so these are quirks we will be figuring out. There will be information in the README but in the program itself it's best to leave it to Qt, which already gives a message.
